### PR TITLE
feat: add utility constructors for ReactiveComponentExecutionMode

### PR DESCRIPTION
**Pull request by nadin-Starkware**
**Stack**:
- #2867
- #2866 ⬅

⚠️ *Part of a stack created by [spr](https://github.com/ejoffe/spr). Do not merge manually using the UI - doing so may have unexpected results.*

### CODE REVIEW

**Review by Itay-Tsabary-Starkware — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/sequencer/2866)* status: 0 of 2 files reviewed, 2 unresolved discussions (waiting on @nadin-Starkware)

___
*[`crates/starknet_sequencer_node/src/config/component_execution_config.rs` line 70 at r1](https://reviewable.io/reviews/starkware-libs/sequencer/2866#-OEij6X1ByeVdAnfqjmX:-OEij6X1ByeVdAnfqjmY:b-k8pc2c) ([raw file](https://github.com/starkware-libs/sequencer/blob/12aed6dcb6b73410d1f2212bc0490033ba1649a5/crates/starknet_sequencer_node/src/config/component_execution_config.rs#L70)):*
> ```XML
>             remote_server_config: None,
>         }
>     }
> ```

Call `local_with_remote_disabled()`



_Code quote:_
```XML
    fn default() -> Self {
        Self {
            execution_mode: ReactiveComponentExecutionMode::LocalExecutionWithRemoteDisabled,
            local_server_config: Some(LocalServerConfig::default()),
            remote_client_config: None,
            remote_server_config: None,
        }
    }
```
___
*[`crates/starknet_sequencer_node/src/config/component_execution_config.rs` line 102 at r1](https://reviewable.io/reviews/starkware-libs/sequencer/2866#-OEijFBW8aOaujUb9XVF:-OEijFBW8aOaujUb9XVG:b-m9747i) ([raw file](https://github.com/starkware-libs/sequencer/blob/12aed6dcb6b73410d1f2212bc0490033ba1649a5/crates/starknet_sequencer_node/src/config/component_execution_config.rs#L102)):*
> ```XML
>             remote_server_config: Some(RemoteServerConfig { socket }),
>         }
>     }
> ```

Please move these to be under the `testing` feature.



_Code quote:_
```XML
impl ReactiveComponentExecutionConfig {
    pub fn disabled() -> Self {
        Self {
            execution_mode: ReactiveComponentExecutionMode::Disabled,
            local_server_config: None,
            remote_client_config: None,
            remote_server_config: None,
        }
    }

    pub fn remote(socket: SocketAddr) -> Self {
        Self {
            execution_mode: ReactiveComponentExecutionMode::Remote,
            local_server_config: None,
            remote_client_config: Some(RemoteClientConfig {
                socket,
                ..RemoteClientConfig::default()
            }),
            remote_server_config: None,
        }
    }

    pub fn local_with_remote_enabled(socket: SocketAddr) -> Self {
        Self {
            execution_mode: ReactiveComponentExecutionMode::LocalExecutionWithRemoteEnabled,
            local_server_config: Some(LocalServerConfig::default()),
            remote_client_config: None,
            remote_server_config: Some(RemoteServerConfig { socket }),
        }
    }
```

File: crates/starknet_integration_tests/src/end_to_end_integration.rs [hunk 1]
```diff
@@ -124,3 +136,37 @@ pub async fn end_to_end_integration(mut tx_generator: MultiAccountTransactionGen
...
+        consensus_manager: ActiveComponentExecutionConfig::disabled(),
+        mempool: ReactiveComponentExecutionConfig::disabled(),
+        mempool_p2p: ReactiveComponentExecutionConfig::disabled(),
+        state_sync: ReactiveComponentExecutionConfig::disabled(),
+        l1_provider: ReactiveComponentExecutionConfig::disabled(),
+    }
+}
+
+#[cfg(any(feature = "testing", test))]
+async fn get_non_http_component_config(gateway_socket: SocketAddr) -> ComponentConfig {
```
[the remark this answers is not in the document]
Comment (nadin-Starkware) [reply crates/starknet_integration_tests/src/end_to_end_integration.rs:156]: Done.

**Review by nadin-Starkware — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/sequencer/2866)* status: 0 of 4 files reviewed, 3 unresolved discussions (waiting on @graphite-app[bot] and @Itay-Tsabary-Starkware)

File: crates/starknet_integration_tests/src/end_to_end_integration.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (Itay-Tsabary-Starkware) [reply crates/starknet_integration_tests/src/end_to_end_integration.rs]: :wat:
The bot's comment is wrong. Please ignore it, revert its request change, and dismiss its review if it's still blocking.

**Review by Itay-Tsabary-Starkware — CHANGES_REQUESTED** [review]
:lgtm:

> Reviewed 3 of 4 files at r2, 1 of 1 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/sequencer/2866#-:-OEmrqzZ49id2XTqNuIE:bnfp4nl)* status: all files reviewed, 4 unresolved discussions (waiting on @graphite-app[bot] and @nadin-Starkware)

___
*[`crates/starknet_integration_tests/Cargo.toml` line 9 at r2](https://reviewable.io/reviews/starkware-libs/sequencer/2866#-OEmrP1XD3Kf2H2Xgzbe:-OEmrP1XD3Kf2H2Xgzbf:b-pjhxjt) ([raw file](https://github.com/starkware-libs/sequencer/blob/c004c339abe23943aa98f374e8588913ed68df95/crates/starknet_integration_tests/Cargo.toml#L9)):*
> ```TOML
> 
> [features]
> testing = []
> ```

Please remove this, see below.



_Code quote:_
```TOML
[features]
testing = []
```
___
*[`crates/starknet_integration_tests/src/end_to_end_integration.rs` line 1 at r2](https://reviewable.io/reviews/starkware-libs/sequencer/2866#-OEmrA7U-Ut3zinzt4bs:-OEmrA7U-Ut3zinzt4bt:bqxmb9n) ([raw file](https://github.com/starkware-libs/sequencer/blob/c004c339abe23943aa98f374e8588913ed68df95/crates/starknet_integration_tests/src/end_to_end_integration.rs#L1)):*
> ```XML
> #[cfg(any(feature = "testing", test))]
> ```

This module is not a library (i.e., other crates do not import from it).
It does not need to define features. Please remove these throughout.



_Code quote:_
```XML
#[cfg(any(feature = "testing", test))]
```

File: crates/starknet_integration_tests/src/end_to_end_integration.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (nadin-Starkware) [reply crates/starknet_integration_tests/src/end_to_end_integration.rs:130]: I want to remove all these pub anyway, I saw its blocking

**Review by nadin-Starkware — COMMENTED** [review]
> Dismissed @graphite-app[bot] from a discussion.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/sequencer/2866)* status: 3 of 4 files reviewed, 2 unresolved discussions (waiting on @Itay-Tsabary-Starkware)

___
*[`crates/starknet_integration_tests/src/end_to_end_integration.rs` line 1 at r2](https://reviewable.io/reviews/starkware-libs/sequencer/2866#-OEmrA7U-Ut3zinzt4bs:-OEnFuw5C1eV1BZ7opGe:b-896fix) ([raw file](https://github.com/starkware-libs/sequencer/blob/c004c339abe23943aa98f374e8588913ed68df95/crates/starknet_integration_tests/src/end_to_end_integration.rs#L1)):*
<details><summary><i>Previously, Itay-Tsabary-Starkware wrote…</i></summary><blockquote>

This module is not a library (i.e., other crates do not import from it).
It does not need to define features. Please remove these throughout.
</blockquote></details>

Done.

**Review by nadin-Starkware — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/sequencer/2866)* status: 3 of 4 files reviewed, 2 unresolved discussions (waiting on @Itay-Tsabary-Starkware)

___
*[`crates/starknet_integration_tests/Cargo.toml` line 9 at r2](https://reviewable.io/reviews/starkware-libs/sequencer/2866#-OEmrP1XD3Kf2H2Xgzbe:-OEnGC7y-At7R8MKRa9-:b-896fix) ([raw file](https://github.com/starkware-libs/sequencer/blob/c004c339abe23943aa98f374e8588913ed68df95/crates/starknet_integration_tests/Cargo.toml#L9)):*
<details><summary><i>Previously, Itay-Tsabary-Starkware wrote…</i></summary><blockquote>

Please remove this, see below.
</blockquote></details>

Done.

**Review by Itay-Tsabary-Starkware — APPROVED** [review]
:lgtm:

> Reviewed 1 of 1 files at r4, 1 of 1 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/sequencer/2866#-:-OEnReH22vRgkoEAR-pU:bnfp4nl)* status: <span>:shipit:</span> complete! all files reviewed, all discussions resolved (waiting on @nadin-Starkware)

**Comment by nadin-Starkware** [comment]
✓ Commit merged in pull request [#2867](https://github.com/starkware-libs/sequencer/pull/2867)

**Pull request closed without merging**